src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2549,19 +2589,103 @@ public Response resetFileStore(@PathParam("identifier") String dvIdtf,
         }
         if (!user.isSuperuser()) {
             return error(Response.Status.FORBIDDEN, "Superusers only.");
-    	}
-        
-        Dataset dataset; 
-        
+        }
+
+        Dataset dataset;
+
         try {
             dataset = findDatasetOrDie(dvIdtf);
         } catch (WrappedResponse ex) {
             return error(Response.Status.NOT_FOUND, "No such dataset");
         }
-        
+
         dataset.setStorageDriverId(null);
         datasetService.merge(dataset);
-    	return ok("Storage reset to default: " + DataAccess.DEFAULT_STORAGE_DRIVER_IDENTIFIER);
+        return ok("Storage reset to default: " + DataAccess.DEFAULT_STORAGE_DRIVER_IDENTIFIER);
+    }
+
+    @GET
+    @Path("{identifier}/curationLabelSet")
+    public Response getCurationLabelSet(@PathParam("identifier") String dvIdtf,
+            @Context UriInfo uriInfo, @Context HttpHeaders headers) throws WrappedResponse {
+
+        Dataset dataset;
+
+        try {
+            dataset = findDatasetOrDie(dvIdtf);
+        } catch (WrappedResponse ex) {
+            return error(Response.Status.NOT_FOUND, "No such dataset");
+        }
+
+        return response(req -> ok(dataset.getEffectiveCurationLabelSetName()));
+    }
+
+    @PUT
+    @Path("{identifier}/curationLabelSet")
+    public Response setCurationLabelSet(@PathParam("identifier") String dvIdtf,
+            String curationLabelSet,
+            @Context UriInfo uriInfo, @Context HttpHeaders headers) throws WrappedResponse {
+
+        // Superuser-only:
+        AuthenticatedUser user;
+        try {
+            user = findAuthenticatedUserOrDie();
+        } catch (WrappedResponse ex) {
+            return error(Response.Status.BAD_REQUEST, "Authentication is required.");
+        }
+        if (!user.isSuperuser()) {
+            return error(Response.Status.FORBIDDEN, "Superusers only.");
+        }
```

Comment:
I'm a little confused. The sets are defined on dataverse collections but only superusers can define the sets? Shouldn't admins of the collections be able to define the sets and otherwise manage them (including deleting them)?

Reply:
@qqmyers and I discussed, and I said I was OK with this for now. We can revisit opening this up later if it's widely adopted and/or becomes a support burden.

Reply:
Fair enough but the docs and release notes could certainly use some clarification on who can do what when. It wasn't until I got the GUI working (thanks, Jim), that it's obvious that curators/admins in the GUI cannot set up the sets. They can only choose from predefined sets:

![Screen Shot 2021-09-27 at 3 31 56 PM](https://user-images.githubusercontent.com/21006/134973837-032e11e8-7d18-43f8-a72b-ba9c4f98cf7a.png)

The curators/admins in the GUI also cannot preview what's in the sets. The sysadmin knows that the following was used to create the sets in this case:

```
{
  "Standard Process": [
    "Author contacted",
    "Privacy Review",
    "Awaiting paper publication",
    "Final Approval"
  ],
  "Alternate Process": [
    "State 1",
    "State 2",
    "State 3"
  ]
}
```

Reply:
Better labels would probably help - e.g. including an org. name that matches the info in their collection and their current curation process. If we also have the use case where people will be selecting an unknown process to start using, then showing the labels, having more documentation about their meaning, flow diagram for allowed transitions, etc. would all be important.

Reply:
Does the model for implementing this require the use of a header? What if it is desirable to have a list of straight-forward, not grouped options, as was shown in the demo?

Reply:
There may be some confusion: The UI in https://github.com/IQSS/dataverse/pull/7967#issuecomment-885191872 doesn't change - only one set of labels shows up there (no header). Only the new UI for selecting which set will show up for a given dataset is where we either show just the set labels (as is, shown above) or we also provide more info about the items available in a set (as suggested). In this new mgmt UI, the only choices are which set to choose - the individual items are not selectable/ are only informational.

Reply:
What if curators need to see the list of what is under the headers to make the correct choice?

Reply:
Hi - I'd be happy with this being managed through the API only, for superusers only, settable at the collection level or installation level. 

If we do want to implement this in the UI at the collection level, we should make it similar to the S3 selection where it can only be seen by superusers. I think the management of this process should be a coordination/negotiation between the superuser(s) of the installation and the admin(s) of a collection, and that it's important for a superuser to control which collections have what labels enabled, especially once we get into interoperation with other systems.

Reply:
I think this is resolved. It was unclear that the Dataverse General Info/Edit API only shows the choices to superusers.

Reply:
Right. When I started this thread I confused, thinking that curators/admins would be selecting the curation label set. From what I understand now, this is done by superusers who have been working closely with sysadmins that have created the sets in the first place. Of course, new superusers will be onboarded over time so I think longer term it would be nice to see the details (the curation steps or states) in the GUI.

Reply:
Thanks all for the discussion here. If this is widely adopted we can revisit by adding more details and/or making it available to other roles than superusers. I'll resolve this conversation.